Bug fix: allow compile with LMIC_USE_INTERRUPTS defined

terrillmoore opened the pull request:
Without this change, you cannot compile the library with #define LMIC_USE_INTERRUPTS in your config. I need interrupts for my project.

terrillmoore committed:
Bug fix: allow compile with LMIC_USE_INTERRUPTS defined

terrillmoore committed:
Add LMIC_getSeqnoUp() API, primarily for debugging.

terrillmoore committed:
Add lmic_project_config.h, change config.h to use it, and update README.md to document

terrillmoore committed:
Fix comment for DISABLE_MCMD_BCNI_ANS

terrillmoore committed:
Merge branch 'feature-lmic_project_config'

frankleonrose commented:
All these changes are now already merged into the master branch, so I think it's safe to close this PR. You could pull arduino-lmic/master and resolve conflicts to confirm.

terrillmoore committed:
Merge in changes from upstream

terrillmoore committed:
Honor the channel mask while trying to join the network. If you know
you're on TTN, then there is no reason to send JOIN packets on non-
TTN channels. If you don't like this behavior, don't set a subband
prior to trying to join.

terrillmoore committed:
Squashed commit: When enabling/disabling subbands, also enable/disable the associated 500 kHz channel

commit 17aa932ba7655402263895496f86a7de53e44298
Merge: 79be46f bf6c45d
Author: Terry Moore <tmm@mcci.com>
Date:   Sun Sep 11 20:24:26 2016 -0400

    Merge changes from master; and deal with switch back to origin 0.

commit 79be46f8ac5869a59c86bb9d7b6d2fb814326cb1
Author: Terry Moore <tmm@mcci.com>
Date:   Sat Sep 10 23:27:37 2016 -0400

    When enabling/disabling subbands, also enable/disable the associated 500 kHz channel

terrillmoore committed:
Don't set up to use interrupts by default in the sample lmic_project_config.h

terrillmoore committed:
Note that we've tested on the Feather M0 LoRa

terrillmoore commented:
I pulled in changes from upstream, merged, and then added changes:
1. honor the channel mask while trying to join. If you've selected a subband, there's really no point in searching all the bands.
2. Change the subband enable/disable code to enable/disable the 500 kHz channel corresponding to the eight 125 kHz channels.
3. Remove an error: I should not have enabled use of interrupts by default.
4. Add the Feather M0 LoRa 900 to the list of tested boards.

In my testing, these changes make joining much faster and consistent.

frankleonrose commented on src/lmic/lmic.c:
```diff
@@ -880,14 +880,26 @@ static void initJoinLoop (void) {
     if( LMIC.datarate != DR_SF8C ) {
+        // assume that 500 kHz equiv of last 125 kHz channel 
+	// is also enabled, and use it next.
         LMIC.txChnl = 64+(LMIC.txChnl&7);
```
@terrillmoore Shouldn't this line be `LMIC.txChnl = 64+(LMIC.txChnl >> 3);`?

Maybe we should define something like this:

```
#define SubBandOfChannel(c) (c<64 ? (c>>3) : (c & 7))
```

frankleonrose reviewed (approved):
Looks good to me. I'll cherry pick what I can and submit upstream.

terrillmoore commented:
You’re right about the error in the comment below. Will you correct or shall I fix this and re-submit?

--Terry

From: Frank Leon Rose [mailto:notifications@github.com] 
Sent: Tuesday, September 27, 2016 09:20
To: things-nyc/arduino-lmic arduino-lmic@noreply.github.com
Cc: Terry Moore tmm@mcci.com; Mention mention@noreply.github.com
Subject: Re: [things-nyc/arduino-lmic] Bug fix: allow compile with LMIC_USE_INTERRUPTS defined (#8)

@frankleonrose approved this pull request.

Looks good to me. I'll cherry pick what I can and submit upstream.

---

In src/lmic/lmic.c https://github.com/things-nyc/arduino-lmic/pull/8#pullrequestreview-1739924 :

> ```
>  //
>  u1_t failed = 0;
>  if( LMIC.datarate != DR_SF8C ) {
> ```
> -        // assume that 500 kHz equiv of last 125 kHz channel 
> -       // is also enabled, and use it next.
>        LMIC.txChnl = 64+(LMIC.txChnl&7);

@terrillmoore https://github.com/terrillmoore  Shouldn't this line be LMIC.txChnl = 64+(LMIC.txChnl >> 3);?

Maybe we should define something like this:

#define SubBandOfChannel(c) (c<64 ? (c>>3) : (c & 7))

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub https://github.com/things-nyc/arduino-lmic/pull/8#pullrequestreview-1739924 , or mute the thread https://github.com/notifications/unsubscribe-auth/ASb_7x1-tb1qcpXQ_Z3D5YO3i9bAm_sZks5quSYNgaJpZM4J5krN .

frankleonrose commented:
@terrillmoore I fixed it before submitting the PR.
